src/streamer.rs

```diff
@@ -442,6 +442,39 @@ fn process_blob(
     }
 }
 
+fn validate_blob_against_window(
+    debug_id: u64,
+    pix: u64,
+    consumed: u64,
+    received: &mut u64,
+) -> bool {
```

Comment:
Return `Option<u64>`?

Reply:
with u64 carrying received?

Reply:
@garious , wondering how Option helps here. Is this to differentiate between different error conditions? Or something more?

Reply:
Yes, with u64 carrying received. It's more idiomatic Rust to avoid `mut`. You'll see those tests get shorter.